earthfile2llb/converter.go

```diff
@@ -1765,6 +1792,45 @@ func (c *Converter) targetInputActiveOnly() dedup.TargetInput {
 	return c.mts.Final.TargetInput().WithFilterBuildArgs(activeBuildArgs)
 }
 
+// persistCacheVolume makes a temporary cache volume permanent by writing it's contents
+// from the volume to the host filesystem at the same directory.
+// Used when the Target contains a `CACHE --persist /my/directory` directive (with the --persist).
+func (c *Converter) persistCacheVolume() {
```

Comment:
Small amount of code, large amount of hassle. The operations in this function are a bit delicate and took a lot of trial and error to get right. 

For example, 
* it seems the cache volume isn't totally unmounted in a successive operation on the image without the run option that includes the volume. An empty directory seems to linger.
* Commands to `mv` or `cp` files to or from the directory where the volume is/was mounted fail for reasons I don't understand
  * e.g. before switching to llb.Copy, I was invoking `cp -r` to move the files to the directory that used to have the cache volume mounted to it - that would fail unless I put them in a sub directory, putting the files in the directory at top-level would fail for some reason

Reply:
Regarding cache volumes not being totally unmounted, this example shows the same behaviour using `RUN --mount=type=cache` :
```Earthfile
mount-test:
    RUN --mount=type=cache,target=/mnt/test echo "hello world" > /mnt/test/hello
    RUN --mount=type=cache,target=/mnt/test cat /mnt/test/hello
    RUN ls /mnt
    RUN ls /mnt/test
```
=>
```
         +mount-test | --> RUN echo "hello world" > /mnt/test/hello
         +mount-test | --> RUN cat /mnt/test/hello
         +mount-test | hello world
         +mount-test | --> RUN ls /mnt
         +mount-test | test
         +mount-test | --> RUN ls /mnt/test
```